lib/new_relic/agent/configuration/default_source.rb

```diff
@@ -1094,6 +1094,14 @@ def self.enforce_fallback(allowed_values: nil, fallback: nil)
           :allowed_from_server => false,
           :description => 'Controls auto-instrumentation of gRPC clients at start up.  May be one of [auto|prepend|chain|disabled].'
         },
+        :'instrumentation.grpc.host_denylist' => {
+          :default => ['tracing\.(staging-)?edge\.nr-data'],
+          :public => true,
+          :type => Array,
+          :allowed_from_server => false,
```

Comment:
Out of curiosity, what goes into the true/false determination for allowing a value to be set from the server?

Reply:
We should document this answer in the file! I've often had that question. My understanding is that it should be allowed_from_server if it can be set via server-side config, or if the value could be manipulated by the collector or another out of agent response.

Reply:
I think the only time we would set this to true would be based on agent specs probably. We would set it to true if we knew that the collector was going to be sending us info for this setting, but most settings the collector doesn't care about or give us back things to use.